File: 정수빈/LeetCode/medium/unique-paths.py
# https://leetcode.com/problems/unique-paths/
# 백트래킹 문제로 0,0에서 해당지점까지 가는 문제로, 메트릭스로 플어도 되지만 간단하게 순열조합으로 풀었습니다. 

class Solution(object):
    def uniquePaths(self, m, n):
        import math
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        return math.factorial(m+n-2)//(math.factorial(m-1) * math.factorial(n-1))
        

File: 장세훈/LeetCode/medium/UniquePaths.java
class UniquePaths {
  /*
      https://leetcode.com/problems/unique-paths/
   */

  public int uniquePaths(int m, int n) {
    int[][] map = new int[m][n];

    for (int i = 0; i < m; i++) {
      map[i][0] = 1;
    }

    for (int i = 0; i < n; i++) {
      map[0][i] = 1;
    }

    for (int i = 1; i < m; i++) {
      for (int j = 1; j < n; j++) {
        map[i][j] = map[i - 1][j] + map[i][j - 1];
      }
    }

    return map[m - 1][n - 1];
  }
}